module-2/backend/week_three.md

```diff
@@ -9,15 +9,40 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 
 ### Questions
 
-1. What is the entry at the command line to create a new rails app?
-2. What do Models generally inherit from in rails?
-3. What do Controllers generally inherit from in a rails project?
-4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
-5. What rake task is useful when looking at routes, and what information does it give you?
-6. What is an example of a route helper? When would you use them?
-7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
-8. What are strong params and why are the necessary?
-9. What role does `form_for` play in helping us create our forms?
-10. How does `form_for` know where to submit the user's input?
+1. What is the entry at the command line to create a new rails app? rails new put_name_here --skip-spring --skip-turbolinks (also can chang database being used here with --database=postgresql)
+
+2. What do Models generally inherit from in rails? Application Record
+
+3. What do Controllers generally inherit from in a rails project? ApplicationController
+
+4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality? resources :horses, only [:show]
+
+5. What rake task is useful when looking at routes, and what information does it give you? rake routes will give you all the URI and controller path names for any given resource that you've specified so that you can call paths in a variety of ways from controllers and views
+6. What is an example of a route helper? When would you use them? these are things that rails "knows" how to call; you can write new_horses_path in a link_to of a view instead of writing the actual path like this '/horses/new'
+
+7. What's the difference between what `_url` and `_path` return when combined with a routes prefix? not sure about this one
+
+8. What are strong params and why are the necessary? strong params tell rails what attributes you're allowing someone to submit and store in the database from a form; it's basically a security feature so that anyone submitting from an online form can't mess up the database
+
+9. What role does `form_for` play in helping us create our forms? form_for is a rails method that tells an application what kind of params are going to be coming forth in an online form, you have control of setting what  fields the form contains, but rails already knows that it's a form that will be used to submit info from a user into the database
+
+10. How does `form_for` know where to submit the user's input? you specify this information in the controller by picking which action calls which erb file. The controller then "knows" which path to submit the form from based on the routes you've defined. 
```

Comment:
`form_for` checks the argument you provide it (e.g. `@horse` in the example below) to see if it's been saved to the database. If it hasn't, `form_for` returns a form that is set to submit a POST request to `/horses`. If it has been saved, `form_for` returns an HTML form that is set to submit a PATCH request to `/horses/:id`.